### DISCUSSION TITLE
How to use cookie in phalcon 4.x

### DISCUSSION THREAD

**Discussion by kinhdang** [phalcon/phalcon#376] [General]
I'm newbie in phalcon, maybe this question is stupid, Please give me why this is not woking propely.

**in my Autoloader**

$container = new FactoryDefault();
.
.
.
.
.
.
$container->set(
    'crypt',
    function () {
        $crypt = new Crypt();
        
        // The `$key' should have been previously generated in a cryptographically safe way.
        $key = "T4\xb1\x8d\xa9\x98\x05\\\x8c\xbe\x1d\x07&[\x99\x18\xa4~Lc1\xbeW\xb3";
        
        $crypt->setKey($key);
        
        return $crypt;
    }
);

$container->set(
    'cookies',
    function () {
        $cookies = new Cookies();
        
        // The `$key' MUST be at least 32 characters long and generated using a
        // cryptographically secure pseudo random generator.
        $key = "#1dj8$=dp?.ak//j1V$~%*0XaK\xb1\x8d\xa9\x98\x054t7w!z%C*F-Jk\x98\x05\\\x5c";
        
        $cookies->setSignKey($key);
        
        return $cookies;
    }
);

**example in Controller:** 
 
public function indexAction()
    {     
        Phalcon\Tag::setTitle('..:: Login'); 
		
        //get cookie 
        var_dump($this->cookies->getCookies()); die();
		/* result when bug value
		but alway return 
		array (size=0)
		   empty
		*/
    }
     
    public function detailviewAction(){
        //set cookie 
        $this->cookies->set('go_result','Y',time()+600, '/');
        $this->cookies->send();
    }

thanks,

**Comment by niden** [comment]
v6 is not ready for release. Please redirect your question to the main repo here:

https://github.com/orgs/phalcon/discussions

Thanks!
